src/plugins/plugin.filler.js

```diff
@@ -190,37 +190,82 @@ function isDrawable(point) {
 	return point && !point.skip;
 }
 
-function drawArea(ctx, curve0, curve1, len0, len1) {
-	var i;
+function drawArea(ctx, curve0, curve1, len0, len1, mode, area) {
+	var i, above, below;
+	var fillAreaPointsSet = [];
+	var clipAreaPointsSet = [];
 
 	if (!len0 || !len1) {
 		return;
 	}
 
+	switch (mode) {
+	case 'above':
+		above = true;
+		break;
+	case 'below':
+		below = true;
+		break;
+	default:
+		above = false;
+		below = false;
+	}
+
+	if (below || above) {
+		if (above) {
+			clipAreaPointsSet.push({x: curve1[len1 - 1].x, y: area.top});
+		} else {
+			clipAreaPointsSet.push({x: curve0[0].x, y: area.top});
+			clipAreaPointsSet.push(curve0[0]);
+		}
+	}
+
 	// building first area curve (normal)
-	ctx.moveTo(curve0[0].x, curve0[0].y);
+	fillAreaPointsSet.push(curve0[0]);
 	for (i = 1; i < len0; ++i) {
-		helpers.canvas.lineTo(ctx, curve0[i - 1], curve0[i]);
+		curve0[i].flip = false;
+		fillAreaPointsSet.push(curve0[i]);
+		if (below) {
+			clipAreaPointsSet.push(curve0[i]);
+		}
 	}
 
 	// joining the two area curves
-	ctx.lineTo(curve1[len1 - 1].x, curve1[len1 - 1].y);
+	curve1[len1 - 1].joint = true;
+	fillAreaPointsSet.push(curve1[len1 - 1]);
 
 	// building opposite area curve (reverse)
 	for (i = len1 - 1; i > 0; --i) {
-		helpers.canvas.lineTo(ctx, curve1[i], curve1[i - 1], true);
+		if (above) {
+			clipAreaPointsSet.push(curve1[i]);
+		}
+		curve1[i - 1].flip = true;
+		fillAreaPointsSet.push(curve1[i - 1]);
+	}
+	if (above || below) {
+		if (above) {
+			clipAreaPointsSet.push(curve1[0]);
+			clipAreaPointsSet.push({x: curve1[0].x, y: area.top});
+		} else {
+			clipAreaPointsSet.push({x: curve0[len0 - 1].x, y: area.top});
+		}
 	}
+	return [clipAreaPointsSet, fillAreaPointsSet];
 }
 
-function doFill(ctx, points, mapper, view, color, loop) {
+function doFill(ctx, points, mapper, view, color, loop, mode, area) {
 	var count = points.length;
 	var span = view.spanGaps;
 	var curve0 = [];
 	var curve1 = [];
 	var len0 = 0;
 	var len1 = 0;
-	var i, ilen, index, p0, p1, d0, d1;
+	var fillAreaPointsSets = [];
+	var clipAreaPointsSets = [];
+	var specMode = mode === 'all' || mode === 'above' || mode === 'below';
```

Comment:
I think this can be removed (everything else than above/below would default to all then.